rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -3,180 +3,397 @@
 #include "Emu/Cell/PPUModule.h"
 
 #include "cellMic.h"
+#include <Emu/IdManager.h>
+#include <Emu/Cell/lv2/sys_event.h>
+#include <thread>
+
+#include "3rdparty/OpenAL/include/al.h"
+#include "3rdparty/OpenAL/include/alc.h"
+#include "3rdparty/OpenAL/include/alext.h"
 
 logs::channel cellMic("cellMic");
 
+void mic_thread::on_init(const std::shared_ptr<void>& _this)
+{
+	// Need to init buffer for relloc to work.
+	buffer = (ALbyte*)malloc(0);
+
+	named_thread::on_init(_this);
+}
+
+void mic_thread::on_task()
+{
+	while (fxm::check<mic_thread>() && !Emu.IsStopped())
+	{
+		if (Emu.IsPaused())
+		{
+			std::this_thread::sleep_for(1ms); // hack
+			continue;
+		}
+		if (!micOpened)
+			continue;
+
+		if (!micStarted)
+			continue;
+
+		if (inputDevice == nullptr)
+		{
+			inputDevice = alcCaptureOpenDevice(NULL, DspFrequency, AL_FORMAT_STEREO_FLOAT32, rawFrequency / 2);
+			alcCaptureStart(inputDevice);
+		}
+
+		// If event queue is not set, then we can't send any events
+		if (eventQueueKey == 0)
+			continue;
+
+		int count = 0;
+		while (count < 1)
+		{
+			alcGetIntegerv(inputDevice, ALC_CAPTURE_SAMPLES, 1, &count);
+			std::this_thread::sleep_for(10ms);
+		}
+		ALuint frameSize = 2 * bitResolution / 8;
+		buffer = (ALbyte *) relloc(frameSize, count);
```

Comment:
should be realloc

Reply:
should also be `realloc(buffer, frameSize * count)` or something like this.